### PR TITLE
fix: extra links shown in boxes

### LINKED ISSUE #6080: Public Events Page Sidebar Link Widget: Show Extra Links in boxes
The sidebar widget for social links already has nice boxes with icons. Please change the extra links (defined in wizard step 2) to display the text in a square box as well, e.g. Blog to show up with text "Blog" in a square box.

Also propose a way how a text area should be displayed that is too long, e.g. shortened with dots... or make the font smaller? What is the best option?

Screenshot from 2020-12-18 23-31-02

### PR DESCRIPTION
**Pull request by nlok5923**
Fixes #6080 

#### Short description of what this resolves:
extra link to be shown in boxes

#### Changes proposed in this pull request:

- extra links will be shown in blue boxes

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by iamareebjamal** [comment]
screenshot

**Comment by nlok5923** [comment]
Sir i am not able to add link on clicking next it says Request failed 

`Request failed: PATCH https://api.eventyay.com/v1/events/3866d3ee`
Screenshot (17)

**Comment by nlok5923** [comment]
Screenshot : 
image

**Comment by nlok5923** [comment]
image

**Comment by maze-runnar** [comment]
> before changes:
> image
> After changes:
> image

this issues says `display the text in a square box as well. Blog to show up with text "Blog" in a **square** box.`.

File: app/templates/components/public/social-links.hbs [hunk 1]
```diff
@@ -19,11 +19,10 @@
   </div>
   {{#each this.socialLinks as |socialLink|}}
     {{#if (eq socialLink.normalizedName 'globe')}}
-      <div class="item">
-        <i class="fitted {{socialLink.normalizedName}} disabled icon"></i>
-        <div class="content">
-          <a href="{{socialLink.link}}" target="_blank" rel="noopener nofollow">{{socialLink.displayName}}</a>
-        </div>
+       <div class="m-1">
+        <a href="{{socialLink.link}}" class="ui icon small blue button" target="_blank" rel="noopener nofollow">
```
Comment (maze-runnar) [inline app/templates/components/public/social-links.hbs:23]: there is no icon so it should be `ui small blue button`

**Comment by nlok5923** [comment]
> > before changes:
> > image
> > After changes:
> > image
> 
> this issues says `display the text in a square box as well. Blog to show up with text "Blog" in a **square** box.`. But I think shwoing square button for text like personal would be big. So what about Making it's height only equal to other buttons? or just a regular size?

Won't the dimension of button depend on the text written in it?

**Comment by maze-runnar** [comment]
I am saying about height, not width

**Comment by nlok5923** [comment]
@maze-runnar for increasing height there's no built in variation in sematic ui library .I had to explicitly change it's default size using id. I think changing it explicitly might not good for other long words .

File: app/templates/components/public/social-links.hbs [hunk 2]
```diff
@@ -19,11 +19,10 @@
...
   {{#each this.socialLinks as |socialLink|}}
     {{#if (eq socialLink.normalizedName 'globe')}}
-      <div class="item">
-        <i class="fitted {{socialLink.normalizedName}} disabled icon"></i>
-        <div class="content">
-          <a href="{{socialLink.link}}" target="_blank" rel="noopener nofollow">{{socialLink.displayName}}</a>
-        </div>
+       <div class="m-1">
+        <a href="{{socialLink.link}}" class="ui small blue button" target="_blank" rel="noopener nofollow">
+          <div style="font-size: .8rem;">{{socialLink.displayName}}</div>
```
Comment (ranjitkshah) [inline app/templates/components/public/social-links.hbs:24]: also add css for long text or u can use textoverflow: ellipsis
like this
image

**Review by ranjitkshah — CHANGES_REQUESTED** [review]

File: app/styles/libs/_helpers.scss [hunk 3]
```diff
@@ -98,3 +98,10 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
 .no-wrap {
   white-space: nowrap;
 }
+
+.extra-link {
```
Comment (daretobedifferent18) [inline app/styles/libs/_helpers.scss:102]: there is already a class defined name ```truncate``` which add '...' if text is long. Pls use this

**Comment by Creatoon** [comment]
In order to maintain the square shape , the other link's first letter can be displayed on the box and if someone hovers on it - a tool tip appears showing it's full name. what say?

**Comment by nlok5923** [comment]
> In order to maintain the square shape , the other link's first letter can be displayed on the box and if someone hovers on it - a tool tip appears showing it's full name. what say?

@iamareebjamal  sir can we do this for this issue. I think this would be good.

**Comment by nlok5923** [comment]
@iamareebjamal please review this pr.

**Comment by daretobedifferent18** [comment]
Pls add the screenshot

**Comment by nlok5923** [comment]
Screenshot after latest commit.
image

**Comment by nlok5923** [comment]
@iamareebjamal @mariobehling sir please review this pull request.

**Review by daretobedifferent18 — COMMENTED** [review]
Screenshot from 2020-12-23 17-02-54
- [ ] Box size should be same
- [x] Link shouldn't come on new line

**Comment by nlok5923** [comment]
for maintaining the same box size i had to alter the default styling of semantic ui using id i think it would not be good.

**Comment by nlok5923** [comment]
changes in latest commit:
image

**Review by daretobedifferent18 — CHANGES_REQUESTED** [review]
Could you implement a way in which when we hover the link box, it shows full link name
It will be better for other users to get to know what is the name of the link incase it is not coming in the box otherwise it is tedious for other to get to know

**Comment by nlok5923** [comment]
screenshot after latest changes :
image

File: app/templates/components/public/social-links.hbs [hunk 4]
```diff
@@ -16,15 +16,16 @@
...
-        <div class="content">
-          <a href="{{socialLink.link}}" target="_blank" rel="noopener nofollow">{{socialLink.displayName}}</a>
-        </div>
+      <div class="m-1">
+        <UiPopup @content={{t socialLink.displayName}}>
+        <a href="{{socialLink.link}}" class="ui icon small blue button" target="_blank" rel="noopener nofollow">
+          <div  class="extra-link truncate" style="font-size: .8rem;">{{socialLink.displayName}}</div>
+        </a>
+        </UiPopup>
       </div>
```
Comment (daretobedifferent18) [inline app/templates/components/public/social-links.hbs:27]:
```suggestion
      <div class="m-1">
        <UiPopup @content={{t socialLink.displayName}}>
          <a href="{{socialLink.link}}" style="width: 49.53333px;" class="ui icon blue big button" target="_blank" rel="noopener nofollow">
            <div  class="truncate" style="font-size: .8rem;">{{socialLink.displayName}}</div>
          </a>
        </UiPopup>
      </div>
```

**Review by daretobedifferent18 — CHANGES_REQUESTED** [review]

File: app/templates/components/public/social-links.hbs (hunk 4 above)
Comment (daretobedifferent18) [reply app/templates/components/public/social-links.hbs:27]: Pls do the following changes

File: app/templates/components/public/social-links.hbs [hunk 5]
```diff
@@ -16,15 +16,16 @@
...
   {{#each this.socialLinks as |socialLink|}}
     {{#if (eq socialLink.normalizedName 'globe')}}
-      <div class="item">
-        <i class="fitted {{socialLink.normalizedName}} disabled icon"></i>
-        <div class="content">
-          <a href="{{socialLink.link}}" target="_blank" rel="noopener nofollow">{{socialLink.displayName}}</a>
-        </div>
+      <div class="m-1">
+        <UiPopup @content={{t socialLink.displayName}}>
+          <a href="{{socialLink.link}}" style="width: 49.53333px;" class="ui icon blue big button" target="_blank" rel="noopener nofollow">
```
Comment (daretobedifferent18) [inline app/templates/components/public/social-links.hbs:23]: remove inline style from here and add class 'extra-link'

File: app/styles/libs/_helpers.scss [hunk 6]
```diff
@@ -98,3 +98,7 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
...
+  width: 2rem;
```
Comment (daretobedifferent18) [inline app/styles/libs/_helpers.scss:103]: change from 2rem -> 49.5333px

Comment (daretobedifferent18) [reply app/styles/libs/_helpers.scss:103]: Update the screenshots

Comment (daretobedifferent18) [inline app/styles/libs/_helpers.scss:103]:
```suggestion
  width: 49.5333px;
```

File: app/templates/components/public/social-links.hbs (hunk 5 above)
Comment (daretobedifferent18) [inline app/templates/components/public/social-links.hbs:23]:
```suggestion
          <a href="{{socialLink.link}}" class="ui icon blue big button extra-link" target="_blank" rel="noopener nofollow">
```

**Comment by daretobedifferent18** [comment]
#### Screenshots
Screenshot from 2020-12-23 20-07-48

**Comment by nlok5923** [comment]
Screenshot after latest commit :
image

**Comment by daretobedifferent18** [comment]
@iamareebjamal I think we are done with this PR, pls let us know if to change any further design part or something else.
I was thinking of just displaying first letter of extra-links and on hover: show full name. 
what say?

**Comment by mariobehling** [comment]
Some events have a box "Event Site". This is shown in two rows. Please display two lines if words are longer here. Simply break words and we test a bit how this feels for a few weekks. Show 5 characters per line or 4 characters and t two dots if words are longer on the second line.

**Comment by nlok5923** [comment]
Screenshot after latest commit :
image
for breaking word in rows and displaying second row word with ellipsis if it is long. sir it is only supported by using vendor prefixes(-webkit) is it fine sir.

**Comment by nlok5923** [comment]
@mariobehling  sir please review the changes.

**Comment by mariobehling** [comment]
The look itself is fine. I will open other issues to follow up on more changes after getting more experience over a few days. I leave the review to tech questions to others. Thanks!

**Review by daretobedifferent18 — COMMENTED** [review]
Pls make these all changes as 
Current | After implementing these changes
----- | -------
Screenshot from 2020-12-24 19-26-07 | Screenshot from 2020-12-24 19-29-20

 
Also change as follow
```css

.d-flex {
  display: flex !important;
}
```

Add class
```css
.contents-center {
  justify-content: center !important;
}
```

**Comment by nlok5923** [comment]
@iamareebjamal  sir please review this pull request.

File: app/styles/libs/_helpers.scss [hunk 7]
```diff
@@ -103,6 +103,24 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
...
 }
 
+.extra-link {
+  width: 49.5333px;
+  height: 46.3px;
+}
+
+.clamp {
+  -webkit-box-orient: vertical;
+  -webkit-line-clamp: 2;
```
Comment (iamareebjamal) [inline app/styles/libs/_helpers.scss:113]: Not required

File: app/styles/libs/_helpers.scss [hunk 8]
```diff
@@ -103,6 +103,24 @@ $spacer-heights: 50 100 200 300 400 500 600 700 800 900;
   white-space: nowrap;
 }
 
+.extra-link {
+  width: 49.5333px;
+  height: 46.3px;
```
Comment (iamareebjamal) [inline app/styles/libs/_helpers.scss:108]: Hardcoded

File: app/styles/libs/_helpers.scss (hunk 7 above)
Comment (nlok5923) [reply app/styles/libs/_helpers.scss:113]: Sir line-clamp do not work without vendor prefixes

Comment (iamareebjamal) [reply app/styles/libs/_helpers.scss:113]: That means it's a very new feature. Think of other solution

**Comment by iamareebjamal** [comment]
Fix lint and tests

**Comment by nlok5923** [comment]
After changes screenshot :
image
i searched for the solution but the solution i got for having ellipsis on second line overflow were not good from responsiveness point of view so currently the latest commit break the long word in two linea.

**Pull request merged**